avatars/avatars.js

```diff
@@ -1475,6 +1481,25 @@ class Avatar {
     }
   }
 
+  setVelocity(timeDiffS, lastPosition, currentPosition, currentQuaternion) {
+    // Set the velocity, which will be considered by the animation controller
+    const positionDiff = localVector.copy(lastPosition)
+      .sub(currentPosition)
+      .divideScalar(Math.max(timeDiffS, 0.001))
+      .multiplyScalar(0.1);
+    localEuler.setFromQuaternion(currentQuaternion, 'YXZ');
+    localEuler.set(0, -(localEuler.y + Math.PI), 0);
+    positionDiff.applyEuler(localEuler);
+    this.velocity.copy(positionDiff).add(this.lastVelocity).divideScalar(2);
```

Comment:
I don't understand how this velocity is computed.

It's the average of these two points?